plasmapy/diagnostics/thomson.py

```diff
@@ -542,6 +555,24 @@ def spectral_density(
     else:
         instr_func_arr = None
 
+    # Valildate notch input
+
+    if notches is not None:
+        notches_unitless = notches.to(u.m).value
+
+        if len(np.shape(notches_unitless)) == 1:
```

Comment:
```if len(np.shape(notches_unitless)) == 1:```

What are you checking here? Is there a more direct way to check it?

Reply:
So my thought was that in most instances, there will only be one notch, so the user can input a 1D array with 2 endpoints. But I also wanted to give the flexibility to include multiple notches, so the user can also input a 2D array which is treated as a list of pairs, where each pair is a single notch. I have some checks here to parse both of those possible input types. I did discover np.ndim the other day, so that might be a more concise way to check for the dimensionality of the input.

Reply:
Yes, that's good! Another option would be to accept either a two element np.array or a list of the same. Then you can do something like 

```python
if isinstance(notches, np.ndarray):
    notches = [notches, ]
elif isinstance(notches, list):
   pass
else:
    ValueError

# Check each element in the list notches to make sure it is a valid notch 
```